Reformat classes to use dependency injection.

st55774 opened the pull request:
This pull request adds Guice dependency injection to the project.
Reformat classes to services that will be transformed into singletons.
Remove static contents and using dependency injections.
Create a service for managing password database files.

ondrejdaCZ committed:
Create PasswordDatabaseService and implementation of JSONPasswordDatabaseService.
Implements of JSONPasswordDatabaseService.openPasswordDatabase and JSONPasswordDatabaseService.savePasswordDatabase.
Refactoring CryptoFile class.

ondrejdaCZ committed:
Add Guice dependency injection.
Reformat all classes to service.
Use dependency injection instead of creating instances and static content.
Create PasswordDatabaseService to manage password database files.

rodi0878 commented on src/main/java/cz/upce/fei/inptp/zz/entity/PasswordDatabase.java:
```diff
@@ -18,20 +18,12 @@
 
-    public PasswordDatabase(File file, String passwd) {
-        this.file = file;
+    public PasswordDatabase(String file, String passwd, List<Password> passwords) {
```
This is wrong, File is prefered Java way of abstracting file/folder paths

st55774 replied:
Ok, I was thinking to use java.nio.Path but a lot of methods still use java.io.File so I thought that String will be better. I will fix.

rodi0878 commented on src/main/java/cz/upce/fei/inptp/zz/injector/InstanceInjector.java:
```diff
@@ -0,0 +1,40 @@
+/**
+ * It provides the dependency injector for project.
+ *
+ * @author Ondřej Chrbolka
```
no authors comments - for this we have commit history

st55774 replied:
Removed.

rodi0878 commented on src/main/java/cz/upce/fei/inptp/zz/service/password/JSONPasswordDatabaseService.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    @Override
+    public PasswordDatabase openPasswordDatabase(String path, final String password) {
```
why final password?

st55774 replied:
It shouldn't. When I was using NetBeans IDE (version 8 I think, not sure). When you used the IDE tool to extract some part of code to a method, some parameters made IDE final... Not sure why (IntelliJ doesn't do that in default). Sometimes I forgot a will make parameters final.

rodi0878 commented on src/main/java/cz/upce/fei/inptp/zz/service/password/JSONPasswordDatabaseService.java:
```diff
@@ -0,0 +1,39 @@
+
+    @Override
+    public PasswordDatabase openPasswordDatabase(String path, final String password) {
+        String content = cryptoService.readFile(new File(path), password);
```
new File(...) this also means that you've chosen wrong abstraction and you should stay with File

st55774 replied:
I will fix.

rodi0878 commented on src/main/java/cz/upce/fei/inptp/zz/service/password/JSONPasswordDatabaseService.java:
```diff
@@ -0,0 +1,39 @@
+    public PasswordDatabase openPasswordDatabase(String path, final String password) {
+        String content = cryptoService.readFile(new File(path), password);
+
+        List<Password> passwordFromJSON = new JSONFileService().fromJson(content);
```
no DI here?

st55774 replied:
I forgot. Too much refactoring in the other classes.

rodi0878 commented on src/main/java/cz/upce/fei/inptp/zz/injector/InstanceInjector.java:
```diff
@@ -0,0 +1,40 @@
+     * @return dependency injector.
+     * */
+    public static Injector injector(){
+        if(injector == null){
```
possible threading problem, do not use lazy initialization

st55774 replied:
Sorry about that. We used this method in OOP. I was searching on the internet how to patch it and I found this link (https://stackoverflow.com/questions/8297705/how-to-implement-thread-safe-lazy-initialization) where the author had the same problem. So I used the correct solution to fix.

rodi0878 reviewed (changes requested):
Fine idea but current state is not ready for merging. Please look at my comments and adapt patch to current code state

ondrejdaCZ committed:
Fix problems mentioned by project leader.
Using java.io.File instead of String path.
Remove author in documentation.
Add DI to JSONPasswordDatabaseService.
Remove lazy initialization.

ondrejdaCZ committed:
Merge branch 'master' into dependency-injector

# Conflicts:
#	src/main/java/cz/upce/fei/inptp/zz/service/crypto/CryptoFileService.java

st55774 commented:
@rodi0878 
thank you for your feedback. I fix all issues you mentioned in the comments and resolved merge conflicts. I hope it is all right now. If no, please let me know.
